### PR TITLE
feature: show spinner when route change

### LINKED ISSUE #3900: [FEATURE] Add Loader
### Description

We can add loader, When users navigate to different routes. So application look more user friendly. When i click on different routes i feel like nothing is happing.

### Screenshots

_No response_

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by prajapati55**
## Fixes Issue

Closes #3900

## Check List (Check all the applicable boxes) 

- [ ] My code follows the code style of this project.
- [ ] My change requires changes to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] All new and existing tests passed.
- [ ] This PR does not contain plagiarized content.
- [ ] The title of my pull request is a short description of the requested changes.

## Note to reviewers

<a href="https://gitpod.io/#https://github.com/EddieHubCommunity/LinkFree/pull/4167"></a>

### CODE REVIEW

**Comment by ChinmayMhatre** [comment]
Hey @prajapati55, Could you add a demo video on this PR. Thank you!

File: components/Spinner/RouteSpinner.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
+export default function RouteSpinner() {
+  return (
+    <div className="flex fixed z-50 top-0 h-[100%] w-[100%] bg-black bg-opacity-50">
+      <div
+        role="status"
+        className="absolute -translate-x-1/2 -translate-y-1/2 top-2/4 left-1/2"
+      >
+        <svg
```
Comment (eddiejaoude) [inline components/Spinner/RouteSpinner.js:8]: What are the `svgs? We have ReactIcons in the project

Comment (prajapati55) [reply components/Spinner/RouteSpinner.js]: use fa-spinner icon 
Screenshot (22)

**Comment by prajapati55** [comment]
> Hey @prajapati55, Could you add a demo video on this PR. Thank you!


https://user-images.githubusercontent.com/35908401/215266294-73b32560-af69-4ffa-a0b7-8ee02caca560.mp4

**Review by rahulpandey70 — CHANGES_REQUESTED** [review]
Look good. But you can improve little bit more, As soon as content fetched spinner should be hidden and also make spinner size bigger.

**Comment by prajapati55** [comment]
> Look good. But you can improve little bit more, As soon as content fetched spinner should be hidden and also make spinner size bigger.

increase the size. adding demo video. check,

https://user-images.githubusercontent.com/35908401/215272450-522015c5-b0a3-4d21-b6ed-c22f0091d781.mp4

 please.

**Review by rahulpandey70 — APPROVED** [review]
LGTM

**Comment by eddiejaoude** [comment]
Thank you for the contribution and great collaboration 👍 

I think this could be tweaked to use the benefits from the new version of NextJS we are using
https://www.youtube.com/watch?v=2o5m1ovfl3c

**Comment by prajapati55** [comment]
> 
I get It. Can I reimplement this using streaming and Suspense?

**Comment by ChinmayMhatre** [comment]
> Thank you for the contribution and great collaboration 👍
> 
> I think this could be tweaked to use the benefits from the new version of NextJS we are using https://www.youtube.com/watch?v=2o5m1ovfl3c

Hey Eddie, I believe this works with the new `app` directory in nextjs, currently Linkfree uses the pages folder. We would have to do a major change to shift these. But I think it would be best if we do this after these features are released as stable releases or it may have some bugs. What do you think?

**Comment by eddiejaoude** [comment]
I think they are released as stable now? or will be soon. We don't want to add code we won't need soon.

@prajapati55 thank you for understanding. I appreciate more work needs to be done as @ChinmayMhatre suggested to use the `app` directory. You do not have to do this, maybe close this pull request and create a new one after we have migrated to the `app` directory structure

**Comment by FrozenHearth** [comment]
Just to pitch in, would a progress bar work better?

https://rstacruz.github.io/nprogress/

This is also quite customizable.

**Comment by prajapati55** [comment]
not required at this point in time.

**Comment by eddiejaoude** [comment]
thank you @prajapati55 for your great work. You are welcome to do the PR when we have the latest version of NextJS setup - but I think this is best done after we catch up on the big open PRs otherwise we will have many conflicts

**Pull request closed without merging**